internal/metamorph/callbacker.go

```diff
@@ -135,23 +140,26 @@ func (p *Callbacker) SendCallback(logger *slog.Logger, tx *store.StoreData) {
 		}
 
 		var request *http.Request
-		request, err = http.NewRequest("POST", tx.CallbackUrl, bytes.NewBuffer(statusBytes))
+		request, err = http.NewRequest("POST", callback.CallbackURL, bytes.NewBuffer(statusBytes))
 		if err != nil {
-			logger.Error("Couldn't marshal status", slog.String("url", tx.CallbackUrl), slog.String("token", tx.CallbackToken), slog.String("hash", tx.Hash.String()), slog.String("err", errors.Join(err, fmt.Errorf("failed to post callback for transaction id %s", tx.Hash)).Error()))
+			logger.Error("Couldn't marshal status", slog.String("url", callback.CallbackURL), slog.String("token", callback.CallbackToken), slog.String("hash", tx.Hash.String()), slog.String("err", errors.Join(err, fmt.Errorf("failed to post callback for transaction id %s", tx.Hash)).Error()))
 			return
 		}
 		request.Header.Set("Content-Type", "application/json; charset=UTF-8")
-		if tx.CallbackToken != "" {
-			request.Header.Set("Authorization", "Bearer "+tx.CallbackToken)
+		if callback.CallbackToken != "" {
+			request.Header.Set("Authorization", "Bearer "+callback.CallbackToken)
 		}
 
 		var response *http.Response
 		response, err = p.httpClient.Do(request)
 		if err != nil {
-			logger.Debug("Couldn't send transaction callback", slog.String("url", tx.CallbackUrl), slog.String("token", tx.CallbackToken), slog.String("hash", tx.Hash.String()), slog.String("err", err.Error()))
+			logger.Debug("Couldn't send transaction callback", slog.String("url", callback.CallbackURL), slog.String("token", callback.CallbackToken), slog.String("hash", tx.Hash.String()), slog.String("err", err.Error()))
 			continue
 		}
-		defer response.Body.Close()
+
+		defer func() {
```

Comment:
Why did you changed one-line `defer response.Body.Close()` into this anonymous function?

Reply:
IDE/linter will not return information about unhandled error

Reply:
Reverted change